src/OutputAreaViewModel.ts

```diff
@@ -0,0 +1,374 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use strict';
+
+import {
+  IChangedArgs, Property
+} from 'phosphor-properties';
+
+import {
+  ISignal, Signal
+} from 'phosphor-signaling';
+
+import {
+  ObservableList
+} from 'phosphor-observablelist';
+
+/**
+* A Mime bundle of data.
+*/
+export interface MimeBundle {
+  [key: string]: string;
+  'application/json'?: any;
+}
+
+/**
+* The base interface for an output view model.
+*/
+export
+class OutputBaseViewModel {
+
+  /**
+  * A signal emitted when state of the output changes.
+  */
+  stateChanged: ISignal<OutputBaseViewModel, IChangedArgs<any>>;
+
+  /**
+  * The output type.
+  */
+  outputType: string; // "execute_result" | "display_data" | "stream" | "error"
```

Comment:
I'd rather use strings, since in TS 1.8, string literals can be typing information.  @sccolbert - what do you think?